src/eval-frame/components/reps/__test_helpers__/reps-test-value-cases.js

```diff
@@ -0,0 +1,283 @@
+  weakSet_empty: new WeakSet(),
+  weakMap_empty: new WeakMap()
+};
+
+// ==================== arrays of base objects
+
+export const arraysOfBaseObjects = {};
+Object.keys(baseObjects).forEach(k => {
+  arraysOfBaseObjects[`arrayOfObj_${k}`] = new Array(1000).fill(baseObjects[k]);
+});
+
+// ==================== composite objects
+
+export const compositeObjects = {
+  object_plainComposite1: {
+    id: 2,
+    name: "An ice sculpture",
+    price: 12.5,
+    tags: ["cold", "ice"],
+    time: new Date("2005-04-03"),
+    description: "a very long description string ... ".repeat(1000),
+    dimensions: {
+      length: 7.0,
+      width: 12.0,
+      height: 9.5
+    },
+    warehouseLocation: {
+      latitude: -78.75,
+      longitude: 20.4
+    }
+  },
+
+  object_plainComposite2: {
+    glossary: {
+      title: "example glossary",
+      GlossDiv: {
+        title: "S",
+        GlossList: {
+          GlossEntry: {
+            ID: "SGML",
+            SortAs: "SGML",
+            GlossTerm: "Standard Generalized Markup Language",
+            Acronym: "SGML",
+            Abbrev: "ISO 8879:1986",
+            GlossDef: {
+              para:
+                "A meta-markup language, used to create markup languages such as DocBook.",
+              GlossSeeAlso: ["GML", "XML"]
+            },
+            GlossSee: "markup"
+          }
+        }
+      }
+    }
+  },
+
+  object_plainComposite3: {
+    a1: 1,
+    a2: "A2",
+    a3: true,
+    a4: undefined,
+    a5: {
+      "a5-1": null,
+      "a5-2": ["a5-2-1", "a5-2-2"],
+      "a5-3": {}
+    },
+    a6: () => {
+      console.log("hello world");
+    },
+    a7: new Date("2005-04-03")
+  }
+};
+
+// ==================== custom classes
+
+class Animal {
+  constructor(name) {
+    this.name = name;
+  }
+  speak() {
+    console.log(`${this.name} makes an animal noise.`);
+  }
+}
+
+class Dog extends Animal {
+  constructor(name) {
+    super(`${name} the dog`);
+  }
+
+  speak() {
+    console.log(`${this.name} makes an animal noise.`);
+  }
+}
+
+export const customClassCases = {
+  class_1: new Animal("Bob"),
+  class_subclass: new Dog("Fido")
+};
+
+// ==================== built-ins
+
+export const builtInObjectCases = {
+  intrinsic_math: Math,
+  intrinsic_json: JSON
+};
+
+// ==================== Promise
+
+export const promiseCases = {
+  promise: new Promise(() => "resolve", () => "reject"),
+  promise_reject: Promise.reject(),
+  promise_resolve: Promise.resolve()
+};
+
+// ==================== tabular
+
+export const rowTableCases = {
+  rowsTable_plainObjects: new Array(100).fill(0).map((x, i) => ({
+    index: i,
+    time: new Date(),
+    id: Math.sin(i)
+  })),
+  rowsTable_compositeObjects: new Array(100).fill(compositeObjects),
+  rowsTable_objectsContainingSimpleTypes: new Array(100).fill(simpleTypes),
+  rowsTable_objectsContainingBaseObjects: new Array(100).fill(baseObjects)
+};
+
+export const rowTableFails = {
+  array_oneEmptyObject: [{}],
```

Comment:
note that for testing purposes we've created a bunch of tablular-ish things that should *not* be represented as tables. not also that `arrayOfBaseObjects_emptyArray` covers the case of an array of empty objects, which we should not show as a table.